src/cmake/__init__.py

```diff
@@ -12,9 +18,22 @@ def __dir__():
     return __all__
 
 
-CMAKE_DATA = os.path.join(os.path.dirname(__file__), 'data')
-
-
+if sys.version_info >= (3, 4):
+    cmake_executable_path = None
+    for script in distribution("cmake").files:
+        if str(script).startswith("cmake/data/bin/cmake"):
+            if sys.version_info < (3, 6):
+                # pre-3.6 behavior is strict
+                resolved_script = script.locate().resolve()
+            else:
+                resolved_script = script.locate().resolve(strict=True)
+            cmake_executable_path = resolved_script.parents[1]
+            break
+    CMAKE_DATA = cmake_executable_path if cmake_executable_path else None
```

Comment:
This adds support for editable install adding a generic way of recovering the `cmake/data` directory.

When support for Python < 3.7 will be dropped, the overall logic could be greatly simplified by directly recovering the path to `ctest`, `cmake` and `cpack` using the distribution API.

@henryiii Did I miss anything ?